In [2]:
import numpy as np
from scipy import stats

**Задача 1**

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания mu с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

In [8]:
alpha = 1 - p
n = 256
mean = 80
p = 0.95
std = 16

In [9]:
t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-1.9693105698491928, 1.9693105698491928)

In [10]:
(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))

(78.0306894301508, 81.9693105698492)

**Задача 2**

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. 
1. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? 
2. Найдите P-значение для данного теста.

In [39]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
alpha = 0.01
n = 10
mean = 200

In [40]:
def statistic(samples):
    return (samples.mean() - mean) / (samples.std(ddof=1) / np.sqrt(samples.shape[0]))

In [41]:
t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-3.24983554401537, 3.2498355440153697)

In [43]:
S = statistic(samples)
S

-1.0651074037450896

In [45]:
p_left = stats.t.cdf(S, df=n - 1)
p_right = 1 - stats.t.cdf(S, df=n - 1)
p_left, p_right

(0.15728330176236427, 0.8427166982376357)

In [46]:
pvalue = 2 * min(p_left, p_right)
pvalue

0.31456660352472854

Ответ: 1 - утверждение верно.

**Задача 3**

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

In [5]:
alpha = 0.025
mean = 17.5
n = 100
D = 4

In [6]:
sigma = np.sqrt(D/n)
sigma

0.2


установим уровень значимости 𝛼

In [7]:
alpha = 0.05


Выбранный уровень значимости $\alpha = 0.025$ означает, что для проверки гипотезы о среднем значении alpha = 17 требуется, чтобы выборочное среднее попадало в правосторонний интервал, который равен $a - 2 \cdot \sigma,  a + 2 \cdot \sigma$ (по правилу 2-х сигм).

Kритическая область лежит правее точки $a + 2 \cdot \sigma $ и если выборочное среднее 17.5 mm попадёт в критическую область, то гипотеза $H_0$ отвергается.

In [8]:
a = 17
sigma = 0.2
(a + 2 * sigma)

17.4

Доверительный интервал $\left( 17.4, \infty \right)$ Гипотеза о том, что шарики для подшипников имеют средний диаметр 17 мм, при выбранном уровне значимости $\alpha = 0.025$ отвергается.

In [9]:
stats.norm.ppf(1 - 0.025, 17, 0.2)

17.39199279690801